homeassistant/components/yeelight/__init__.py

```diff
@@ -337,6 +355,11 @@ def setup(self):
         """Fetch initial device properties."""
         initial_update = self._update_properties()
 
-        # We can build correct class anyway.
-        if not initial_update and self.model:
+        if initial_update:
+            # Try to get capabilities for reported model / fw version
+            self._get_capabilities()
             self._initialize_device()
+        else:
+            # We can build correct class anyway.
+            if self.model:
```

Comment:
Should the `self.model` be always set when it goes to this part? What's the behavior if it's non-true?

Reply:
Model wont be set / available from device, if device is offline here. If its offline, but model is forced, we can still initialize correct device.

Reply:
As we have unique_id and device registry, this information will be available any case (to my understanding) after the first initial initialization, no?

Reply:
You are right, it doesn't seem to be needed anymore.

Reply:
Pushed update removing this.